azblob/zt_sas_identity_test.go

```diff
@@ -0,0 +1,85 @@
+package azblob_test
+
+import (
+	"bytes"
+	"fmt"
+	"github.com/Azure/azure-storage-blob-go/azblob"
+	chk "gopkg.in/check.v1"
+	"strings"
+	"time"
+)
+
+func (s *aztestsSuite) TestIdentitySASUsage(c *chk.C) {
+	bsu := getBSU()
+	containerURL, containerName := getContainerURL(c, bsu)
+	blobURL, blobName := getBlockBlobURL(c, containerURL)
+
+	currentTime := time.Now().UTC()
+	accountName, _ := accountInfo()
+
+	ocred, err := getOAuthCredential("")
+	if err != nil {
+		c.Fatal(err)
+	}
+
+	p := azblob.NewPipeline(*ocred, azblob.PipelineOptions{})
+	bsu = azblob.NewServiceURL(bsu.URL(), p)
+
+	keyInfo := azblob.NewKeyInfo(currentTime, currentTime.Add(48*time.Hour))
+
+	budk, err := bsu.GetUserDelegationKey(ctx, keyInfo, nil, nil) //MUST have TokenCredential
+	if err != nil {
+		c.Fatal(err)
+	}
+
+	bSAS, err := azblob.BlobSASSignatureValues{
+		Protocol:          azblob.SASProtocolHTTPS,
+		StartTime:         currentTime,
+		ExpiryTime:        currentTime.Add(24 * time.Hour),
+		Permissions:       "r",
+		ContainerName:     containerName,
+		BlobName:          blobName,
+		UserDelegationKey: budk,
+	}.NewIdentitySASQueryParameters(accountName)
+	if err != nil {
+		c.Fatal(err)
+	}
+
+	//Create pipeline
+	p = azblob.NewPipeline(azblob.NewAnonymousCredential(), azblob.PipelineOptions{})
+
+	//Create blob SAS Identity URL
+	bSASParts := azblob.NewBlobURLParts(blobURL.URL())
+	bSASParts.SAS = bSAS
+	bsurl := bSASParts.URL()
+	fmt.Println(bsurl.String())
```

Comment:
please avoid print statements in tests.

Reply:
That was leftover debug material. Thanks for catching it.